static/app/views/settings/project/filtersAndSampling/filtersAndSampling.tsx

```diff
@@ -87,7 +88,12 @@ class FiltersAndSampling extends AsyncView<Props, State> {
         rule.type === DynamicSamplingRuleType.TRACE
     );
 
-    this.setState({rules: transactionRules});
+    const [rulesWithoutConditions, rulesWithConditions] = partition(
+      transactionRules,
+      transactionRule => !transactionRule.condition.inner.length
+    );
```

Comment:
Do we need to partition them? On line 96 we are just combining them back again and they are not used anywhere else.

Reply:
yes, we do, because since we were  using dynamic sampling before and rules can maybe not be ordered the way we expect, this logic would solve the problem, but ideally would be to know who has dynamic sampling configured and run some backend script to update the order of the objects in the array